### PR TITLE
Added a handler for simple client login/redirection.

### PR DESCRIPTION
**Pull request by benwarfield-usds**
Added a controller to facilitate client login: if the client redirects to it with a known-good URL (based on the allowed CORS origins), redirect back to that URL after login.

### CODE REVIEW

File: src/main/java/gov/usds/case_issues/controllers/LoginRedirectController.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+@Validated
+public class LoginRedirectController {
+
+	private static final Logger LOG = LoggerFactory.getLogger(LoginRedirectController.class);
+
+	@Autowired
+	private WebConfigurationProperties config;
+
+	@GetMapping("/clientLogin")
+	public ResponseEntity<String> shenanigans(@RequestParam(required=false) @URL String redirect) {
```
Comment (timothybest-usds) [inline src/main/java/gov/usds/case_issues/controllers/LoginRedirectController.java]: `shenanigans`?

Comment (benwarfield-usds) [reply src/main/java/gov/usds/case_issues/controllers/LoginRedirectController.java]: Fair.

File: src/main/java/gov/usds/case_issues/controllers/LoginRedirectController.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+		if (redirect == null) {
+			LOG.debug("No redirect provided: defaulting to {}", corsOrigins[0]);
+			validatedRedirect = corsOrigins[0];
+		} else {
+			for (String allowedOrigin : corsOrigins) {
+				LOG.debug("Checking {} against {}", redirect, allowedOrigin);
+				// prevent "http://good.origin:shenanigans/
+				String allowedPrefix = allowedOrigin.endsWith("/") ? allowedOrigin : allowedOrigin + "/";
+				if (redirect.equals(allowedOrigin) || redirect.startsWith(allowedPrefix)) {
+				//if (redirect.startsWith(allowedOrigin)) {
```
Comment (benwarfield-usds) [inline src/main/java/gov/usds/case_issues/controllers/LoginRedirectController.java]: <facepalm>

Comment (benwarfield-usds) [inline src/main/java/gov/usds/case_issues/controllers/LoginRedirectController.java]:
```suggestion
```

Just delete that.

**Comment by nickscialli-usds** [comment]
If possible, it would be helpful to get a `401` response from any non-clientLogin endpoint when not authenticated. That error code would be much more natural to interpret than trying to decode a failed redirect.

**Comment by benwarfield-usds** [comment]
Nick's point is solid, but is a general security-configuration issue that doesn't touch this specific endpoint: going to merge this and make that a separate issue.

**Comment by benwarfield-usds** [comment]
Follow-up issue is #30.

**Pull request merged**